### PR TITLE
ci: support Vulkan build on Windows

### PR DESCRIPTION
**Pull request by vgf89**
I tried compiling the Vulkan feature on Windows, but was running into the vulkan library not being found by the build script. This is the fix. Now I can run tabby on my AMD 5700XT on Windows. It's seems quite fast too!

Changes:
* Remove `=native` in vulkan feature in build.rs so that all library kinds are part of the search. This *shouldn't* break the Linux builds, but if it does CI should catch it.
* Add a CI step to create a Windows Vulkan build automatically. If I'm lucky it'll work first try? 🤞
* User-facing documentation changes to reflect Vulkan support.

### CODE REVIEW

**Comment by vgf89** [comment]
Ah, yeah I had to copy the vulkan-1.lib to vulkan.lib. I need to see if I can fix issue build-side instead of manually

**Comment by vgf89** [comment]
It looks like the vulkan build is currently broken on Linux due to an upstream issue with the manylinux docker container. It's trying to pull the completely wrong (outdated?) URL format for the vulkan sdk.

**Comment by wsxiaoys** [comment]
cc @icycodes

**Comment by icycodes** [comment]
> It looks like the vulkan build is currently broken on Linux CI due to an upstream issue with the manylinux docker container. It's trying to pull the completely wrong (outdated?) URL format for the vulkan sdk.

@vgf89  Thanks, I confirmed that the artifact built by the windows pipeline is working well on my windows machine.   
Looking into the issue on the Linux CI pipeline.

**Comment by icycodes** [comment]
> > It looks like the vulkan build is currently broken on Linux CI due to an upstream issue with the manylinux docker container. It's trying to pull the completely wrong (outdated?) URL format for the vulkan sdk.
> 
> @vgf89 Thanks, I confirmed that the artifact built by the windows pipeline is working well on my windows machine. Looking into the issue on the Linux CI pipeline.

@vgf89 @wsxiaoys 
It appears that the SDK version used in the Linux pipeline has recently been removed from https://vulkan.lunarg.com/sdk/home. The cached download file was being used before this PR. 
I believe this PR can be safely merged without considering the failed Linux CI pipeline. I will address this in the next PR.

File: website/docs/installation/windows/index.mdx
Comment (wsxiaoys) [inline website/docs/installation/windows/index.mdx:1]: let's revert the change of this file since it won't be released until next ver (0.10)

**Review by wsxiaoys — APPROVED** [review]
Otherwise LGTM
